clearblade/cloud/iot_v1/device_types.py

```diff
@@ -37,14 +68,59 @@ def __init__(self, id: str, num_id: str = None,
 
     @staticmethod
     def from_json(json):
-        return Device(id=json['id'], num_id=json['numId'],
-                      credentials=json['credentials'], last_heartbeat_time=json['lastHeartbeatTime'],
-                      last_event_time=json['lastEventTime'], last_state_time=json['lastStateTime'],
-                      last_config_ack_time=json['lastConfigAckTime'], last_config_send_time=json['lastConfigSendTime'],
-                      blocked=json['blocked'], last_error_time=json['lastErrorTime'],
-                      last_error_status_code=json['lastErrorStatus'], config=json['config'],
-                      state=json['state'], log_level=json['logLevel'], meta_data=json['metadata'],
-                      gateway_config=json['gatewayConfig'])
+        lastHeartbeatTimeFromJson = get_value(json,'lastHeartbeatTime')
+        lastEventTimeFromJson = get_value(json, 'lastEventTime')
+        lastStateTimeFromJson = get_value(json, 'lastStateTime')
+        lastConfigAckTimeFromJson = get_value(json, 'lastConfigAckTime')
+        lastConfigSendTimeFromJson = get_value(json, 'lastConfigSendTime')
+        lastErrorTimeFromJson = get_value(json, 'lastErrorTime')
+        
+        convert_times_to_datetime_with_nanoseconds = (False if os.environ.get("TIME_FORMAT") == None else os.environ.get("TIME_FORMAT").lower() == "datetimewithnanoseconds")
+        if convert_times_to_datetime_with_nanoseconds:
+            last_heartbeat_time = None if lastHeartbeatTimeFromJson in [None, ""] else DatetimeWithNanoseconds.from_rfc3339(lastHeartbeatTimeFromJson)
+            last_event_time = None if lastEventTimeFromJson in [None, ""] else DatetimeWithNanoseconds.from_rfc3339(lastEventTimeFromJson)
+            last_state_time = None if lastStateTimeFromJson in [None, ""] else DatetimeWithNanoseconds.from_rfc3339(lastStateTimeFromJson)
+            last_config_ack_time = None if lastConfigAckTimeFromJson in [None, ""] else DatetimeWithNanoseconds.from_rfc3339(lastConfigAckTimeFromJson)
+            last_config_send_time = None if lastConfigSendTimeFromJson in [None, ""] else DatetimeWithNanoseconds.from_rfc3339(lastConfigSendTimeFromJson)
+            last_error_time = None if lastErrorTimeFromJson in [None, ""] else DatetimeWithNanoseconds.from_rfc3339(lastErrorTimeFromJson) 
+        else:
+            last_heartbeat_time = lastHeartbeatTimeFromJson
+            last_event_time = lastEventTimeFromJson
+            last_state_time = lastStateTimeFromJson
+            last_config_ack_time = lastConfigAckTimeFromJson
+            last_config_send_time = lastConfigSendTimeFromJson
+            last_error_time = lastErrorTimeFromJson
+
+        convert_binarydata_to_bytes = (False if os.environ.get("BINARYDATA_FORMAT") == None else os.environ.get("BINARYDATA_FORMAT").lower() == "bytes")
```

Comment:
Just putting my thoughts, is there any way to use inbuilt logic for converting time into Seconds and nanos, instead of adding an additional library of Google to do that?

Reply:
@rajasd27 just realized I don't need to do any conversion on 'config'. I just removed the line that set **convert_binarydata_to_bytes**. It is not needed here.